src/server/game/Spells/Spell.cpp

```diff
@@ -3449,16 +3463,31 @@ void Spell::_cast(bool skipCheck)
 template <class Container>
 void Spell::DoProcessTargetContainer(Container& targetContainer)
 {
+    std::vector<TargetInfoBase*> hitTargets;
     for (TargetInfoBase& target : targetContainer)
-        target.PreprocessTarget(this);
+    {
+        if (!target.MissedTarget(this))
```

Comment:
IMO this shouldnt be done here, isnt target selection a better place?

Reply:
The target selection code is unfamiliar to me which is why I made the change here. Does target selection take into account the delay so it can check only when the projectile lands?

Reply:
Depends if the target needs a miss feedback or not